sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -3,14 +3,97 @@
 
 package com.azure.core.amqp.models;
 
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
 /**
- * Interface representing Amqp Message Body.
+ * This class encapsulates the body of a message. The {@link AmqpBodyType} map to AMQP specification message body types.
+ * Current implementation only support {@link AmqpBodyType#DATA DATA}. Other types will be supported in future releases.
+ * Client should test for {@link AmqpBodyType} before calling corresponding get method.
+ * Get methods not corresponding to the type of the body throw exception.
+ *
+ * @see AmqpBodyType
  */
-public interface AmqpMessageBody {
+public final class AmqpMessageBody {
+    private static final ClientLogger LOGGER = new ClientLogger(AmqpMessageBody.class);
+    private AmqpBodyType bodyType;
+
+    private byte[] data;
+
+    /**
+     * Creates instance of {@link AmqpMessageBody} with given {@link Iterable} of byte array. Please note that this
+     * version of the SDK supports only one element in given {@link Iterable}.
```

Comment:
I guess we should make this javadoc crisp on why we do this. For the customer, this decision seems arbitrary and we need to document the justification for doing this. This, as we discussed earlier, has the potential for dropping data that is sent from another client that does support more than one byte[].